py3status/modules/sysdata.py

```diff
@@ -24,6 +24,9 @@
     zone: Either a path in sysfs to CPU temperature sensor, or an lm_sensors thermal zone to use.
         If None try to guess CPU temperature
         (default None)
+    zone_sensor: Regular expression for capturing specific lm_sensors thermal zone sensor value.
+        Must have only one capturing group.
+        (default r"(?:Core 0|CPU Temp).+\+(.+).+\(.+" )
```

Comment:
(Opinion) I think it would be better not to introduce a new config... so either hardcode the new regex, or having users trying the `lm_sensors` module... mainly because we could make do without `zone` and `zone_sensor` configs if we know what sensors to look for with `cpu_temp`.

In short, having this possible would mean I can assign fan speed to `{cpu_temp}`.

Reply:
Well, yes, this does allow user to assign any sensor to `{cpu_temp}`, which is not ideal.

About removing `zone` - i think that it's possible.
On a generic single desktop cpu system only `coretemp-isa-0000` or `k10temp-pci-00c3` should be present.
Threadripper/EPYC cpus show one k10-temp sensor per die and no generic 'cpu temp' reading, so you'll want to use `lm_sensors` module for them anyway.

About zone_regex:
The generic "all Intel cpus that `coretemp` supports and all Ryzen cpus" regexp may look like this:
'(Package id 0|Core 0|Tdie):\s+\+(\d+\.\d+)'
Works on:
atom d2500 - only core0/1 temps are reported
mobile intel core i3-7 cpus (3-6 gen at least): package and core temps are reported
Ryzen 1/1.5 (Tdie is present).
Ryzen 2: should also have Tdie reported (as well as Tccds) - https://lore.kernel.org/lkml/20200122160800.12560-1-linux@roeck-us.net/

That means having to check which sensor is present during init. I'll experiment with it and make a separate pull request later.

Reply:
By the way, current `sysdata` module will try to open() and readline() anything assigned to `zone`. So nothing is preventing you from assigning arbitrary float-convertible string to `{cpu_temp}` right now.

BTW, I did make 'cpu temp guessing' use predefined sensors. That's just a mockup though, and it will not work for something non-intel/amd (arm, etc).

```
diff --git a/py3status/modules/sysdata.py b/py3status/modules/sysdata.py
index 4f280e71..c3b6e98b 100644
--- a/py3status/modules/sysdata.py
+++ b/py3status/modules/sysdata.py
@@ -99,6 +99,7 @@ SAMPLE OUTPUT
 """
 
 from fnmatch import fnmatch
+from json import loads
 from os import getloadavg
 
 import re
@@ -374,9 +375,9 @@ class Py3status:
                 pass
         if not sensors:
             sensors = self.py3.command_output(command)
-        m = re.search(r"(Core 0|CPU Temp).+\+(.+).+\(.+", sensors)
+        m = re.search(r"(?:Package id 0|Core 0|Tdie):\s+\+(\d+.\d+)", sensors)
         if m:
-            cpu_temp = float(m.groups()[1].strip()[:-2])
+            cpu_temp = float(m.groups()[0].strip())
         else:
             cpu_temp = "?"
 
@@ -437,6 +438,14 @@ class Py3status:
                 sys["format_cpu"] = format_cpu
 
         if self.init["cpu_temp"]:
+            sensors_to_find = ['coretemp-isa-0000', 'k10temp-pci-00c3']
+            sensors_dict = loads(self.py3.command_output(['sensors', '-j']))
+
+            for sensor in sensors_dict:
+                if sensor in sensors_to_find:
+                    self.zone = sensor
+                    break
+
             sys["cpu_temp"] = self._get_cputemp(self.zone, self.temp_unit)
 
         if self.init["load"]:

```

I don't think that half-baked functionality should be left as-is. IMHO, it should either be made as generic as possible (zone and zone_sensor, for example), or removed completely (no cpu_temp in sysdata). Currently you either try to use hwmonX - which reorder themselves every boot, and pinning them with udev is certainly overkill - or use sensor name, but default regex only works for some intel cpus. And i doubt that we can make a regex that encompasses all cpus that py3status can be run at.

Reply:
Resolving this since i will create a new pull-request with updated regex and sensors auto-discovery.